### ISSUE TITLE
while_let_loop with statements false positive

### ISSUE THREAD

**Issue by A4-Tacks** [rust-lang/rust-clippy#16378]
```rust
fn value() -> Option<i32> { None }
fn log(_: &str) { }
pub fn foo() -> i32 {
    loop {
        let Some(2) = value() else {
            log("fail");
            break
        };
        if value().is_some() {
            break;
        }
    }
    3
}
```

**Current output**:

```text
warning: this loop could be written as a `while let` loop
  --> src/main.rs:4:5
   |
 4 | /     loop {
 5 | |         let Some(2) = value() else {
 6 | |             log("fail");
 7 | |             break
...  |
12 | |     }
   | |_____^ help: try: `while let Some(2) = value() { .. }`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#while_let_loop
   = note: `#[warn(clippy::while_let_loop)]` on by default
```

Due to the presence of `log()` and other `break` in loop, it cannot be written as `while let`

**Expect output**:

No warnings

**Other case**:

Add a semicolon after `break`:

```rust
fn value() -> Option<i32> { None }
fn log(_: &str) { }
pub fn foo() -> i32 {
    loop {
        let Some(2) = value() else {
            log("fail");
            break;
        };
        if value().is_some() {
            break;
        }
    }
    3
}
```

This is a false-negative

@rustbot label +C-bug,I-false-positive
